tests/CsrfMiddlewareTest.php

```diff
@@ -165,18 +167,25 @@ private function createStorageMock(string $returnToken): CsrfTokenStorageInterfa
         $mock = $this->createMock(MockCsrfTokenStorage::class);
 
         $mock
-            ->expects($this->once())
             ->method('get')
             ->willReturn($returnToken);
 
         return $mock;
     }
 
+    private function createCsrfMiddleware(bool $autoGenerateToken): CsrfMiddleware
```

Comment:
The test class could be made abstract with this and the following method marked as abstract protected. Then two test cases could be created, one for stateful token and another for stateless token.

Reply:
Done